extensions/assets-manager/AssetsManagerEx.cpp

```diff
@@ -138,22 +170,42 @@ void AssetsManagerEx::prepareLocalManifest()
 
 void AssetsManagerEx::loadLocalManifest(const std::string& manifestUrl)
 {
-    // Prefer to use the cached manifest file, if not found use user configured manifest file
+    Manifest *cachedManifest = nullptr;
+    // Find the cached manifest file
     if (_fileUtils->isFileExist(_cacheManifestPath))
     {
-        _localManifest->parse(_cacheManifestPath);
-        if (_localManifest->isLoaded())
-            prepareLocalManifest();
-        else
-            _fileUtils->removeFile(_cacheManifestPath);
+        cachedManifest = new (std::nothrow) Manifest();
+        if (cachedManifest) {
+            cachedManifest->parse(_cacheManifestPath);
+            if (!cachedManifest->isLoaded())
+            {
+                _fileUtils->removeFile(_cacheManifestPath);
+                CC_SAFE_RELEASE(cachedManifest);
+                cachedManifest = nullptr;
+            }
+        }
     }
     
-    // Fail to found or load cached manifest file
-    if (!_localManifest->isLoaded())
+    // Load local manifest in app package
+    _localManifest->parse(_manifestUrl);
+    if (_localManifest->isLoaded())
     {
-        _localManifest->parse(_manifestUrl);
-        if (_localManifest->isLoaded())
-            prepareLocalManifest();
+        // Compare with cached manifest to determine which one to use
+        if (cachedManifest) {
+            if (strcmp(_localManifest->getVersion().c_str(), cachedManifest->getVersion().c_str()) > 0)
+            {
+                // Recreate storage, to empty the content
+                _fileUtils->removeDirectory(_storagePath);
+                _fileUtils->createDirectory(_storagePath);
```

Comment:
I set my storage path to "./" With the combination of what I just commented above, it ended up removing entire userdata after manifest update. (update from 1.0.9 to 1.0.10)

Ideally this should check the diff between localManifest and cachedManifest to selectively remove unnecessary assets, what do you think?